lib/src/components/NxTag/types.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import {ReactNode, WeakValidationMap, HTMLAttributes} from 'react';
+import * as PropTypes from 'prop-types';
+
+export const NX_TAG_COLORS =
+    ['light-blue', 'purple', 'pink', 'blue', 'red', 'green', 'orange', 'yellow', 'lime', 'indigo'] as const;
+export type NX_TAG_COLORS_TYPE = (typeof NX_TAG_COLORS)[number]; // See https://stackoverflow.com/a/45486495
+
+export interface Props extends Omit<HTMLAttributes<HTMLDivElement>, 'onSelect'> {
+  children: ReactNode;
+  tagColor?: NX_TAG_COLORS_TYPE;
+}
+
+export interface SelectableProps extends Props {
+  onSelect?: (() => void) | null;
+  selected?: boolean;
```

Comment:
Should these really be optional? If someone is using a selectable tag I'd think they must use these props in order for it to work right.

Reply:
> Should these really be optional? If someone is using a selectable tag I'd think they must use these props in order for it to work right.

https://github.com/sonatype/sonatype-react-shared-components/pull/290/commits/9b1783df29bac5384c4c0fe8e29a8efdc1f81829